src/Stratis.Bitcoin.Features.Consensus/Rules/CommonRules/PosCoinviewRule.cs

```diff
@@ -37,20 +37,18 @@ public override void Initialize()
         }
 
         /// <inheritdoc />
-        public override Task RunAsync(RuleContext context)
+        public override async Task RunAsync(RuleContext context)
         {
             this.Logger.LogTrace("()");
 
             // Compute and store the stake proofs.
             this.CheckAndComputeStake(context);
 
-            base.RunAsync(context);
+            await base.RunAsync(context);
 
-            this.stakeChain.Set(context.BlockValidationContext.ChainedHeader, context.Stake.BlockStake);
+            await this.stakeChain.SetAsync(context.BlockValidationContext.ChainedHeader, context.Stake.BlockStake);
```

Comment:
In both cases, please add configure await with false

Reply:
Yes, good catch, my bad.